src/wallet/mod.rs

```diff
@@ -389,6 +387,62 @@ where
         Ok(new_addresses_cached)
     }
 
+    /// Return vector of n unused addresses from the [`KeychainKind`].
+    /// If less than n unused addresses are returned, the rest will be populated by new addresses.
+    /// The unused addresses returned are in order of oldest in keychain first, with increasing index.
+    pub fn get_batch_unused_addresses(
+        &self,
+        n: usize,
+        from_front: bool,
+        keychain: KeychainKind,
+    ) -> Result<Vec<AddressInfo>, Error> {
```

Comment:
I am not feeling comfortable with the API. `get_batched_unused` should not be concerned with `fornt` or `back`. Thats an impl detail for `LastUnused` or `FirstUnused`. And should not be exposed in public API..

This is also breaking the doc above. The order is not maintained anymore.. 

Better to handle the handle the first or last logic in in their respective functions itself than to handle in the batch function which is more generic.